airflow/models/taskinstance.py

```diff
@@ -2571,6 +2571,7 @@ def __eq__(self, other):
         return NotImplemented
 
     def as_dict(self):
+        warnings.warn("This method is deprecated.  Use BaseSerialization.serialize.", DeprecationWarning)
```

Comment:
Probably should have `stacklevel=2` to make the warning show the previous stack?